### PR TITLE
Move SchedulerDelegate(MountingManager) updates to native thread

### PR DESCRIPTION
**Pull request by pprabhu-ng**
1. ProcessMutations and handleCommand is moved to be executed in main thread
2. If transaction update in ongoing, set follow up required flag , which will be used to process the mutation with latest transaction before commit.
3. Commit updates for rendering with latest transaction updates
4. Use TelemetryController interface for pullTransaction updates

### CODE REVIEW

File: ReactSkia/MountingManager.cpp [hunk 1]
```diff
@@ -28,22 +29,19 @@ void MountingManager::BindSurface(RSkSurfaceWindow *surface) {
...
+  RNS_LOG_DEBUG("[" << std::this_thread::get_id() << "] schedulerDidFinishTransaction");
+  /* If transaction processing ongoing,set flag for followUp and return*/
+  if(transactionInFlight_){
+    followUpTransactionRequired_ = true;
     return;
   }
 
-  // auto telemetry = transaction->getTelemetry();
-  // auto number = transaction->getNumber();
+  auto mountingCordinatorCopy = mountingCoordinator;
```
Comment (munezbn) [inline ReactSkia/MountingManager.cpp]: Do we actually need this extra copy ? Looks like  mountingCoordinator is a  shared pointer reference. May be we can reuse ?

Comment (pprabhu-ng) [inline ReactSkia/MountingManager.cpp]: set the transaction flag here

File: ReactSkia/MountingManager.cpp [hunk 2]
```diff
@@ -120,14 +149,18 @@ void MountingManager::ProcessMutations(
...
-  static double prevTime = SkTime::GetMSecs();
-  RNS_LOG_INFO_EVERY_N(60, "Calling Compositor Commit(" << std::this_thread::get_id()) << ") : after " << SkTime::GetMSecs() - prevTime << " ms";
-  prevTime = SkTime::GetMSecs();
+    static double prevTime = SkTime::GetMSecs();
+    RNS_LOG_INFO_EVERY_N(60, "Calling Compositor Commit(" << std::this_thread::get_id()) << ") : after " << SkTime::GetMSecs() - prevTime << " ms";
+    prevTime = SkTime::GetMSecs();
 #endif
-  nativeRenderDelegate_.commit();
-}
+    nativeRenderDelegate_.commitNow();
```
Comment (pprabhu-ng) [inline ReactSkia/MountingManager.cpp]: reuse exiting commit code with usage of flag as arg

File: ReactSkia/MountingManager.cpp [hunk 3]
```diff
@@ -120,14 +149,18 @@ void MountingManager::ProcessMutations(
     }
   }
 
+  if(!followUpTransactionRequired_ ) {
```
Comment (pprabhu-ng) [inline ReactSkia/MountingManager.cpp]: check to be removed

File: ReactSkia/MountingManager.cpp [hunk 4]
```diff
@@ -77,10 +79,37 @@ void MountingManager::schedulerDidClearJSResponder() {
...
+
+}
+
+
 void MountingManager::ProcessMutations(
     ShadowViewMutationList const &mutations,
     SurfaceId surfaceId) {
 
+  RNS_LOG_DEBUG("[" << std::this_thread::get_id() << "] ProcessMutations mutations[" << mutations.size() <<"]");
+
```
Comment (pprabhu-ng) [inline ReactSkia/MountingManager.cpp:115]: check for empty mutation

**Review by munezbn — APPROVED** [review]

**Review by chatrapathi06 — APPROVED** [review]

**Pull request merged**